docs(core): add CI terms to glossary (#22677)
nrwl/nx: docs/shared/reference/glossary.md

> See: [Applications and Libraries](/concepts/more-concepts/applications-and-libraries)

### Atomizer

> The Atomizer automatically splits tasks so that they can be executed in parallel.

Author: Atomizer currently only works on e2e tasks, but Jason just recently mentioned that we plan to expand it to Jest and Vitest.

> The Atomizer automatically splits tasks so that they can be executed in parallel.

Reviewer: Atomizer is natively supported by Nx Cloud Agents, providing you detailed task information in the UI, including interpreted logs and artifacts, automatically grouped by project and workflow.

> See: [Applications and Libraries](/concepts/more-concepts/applications-and-libraries)

### Atomizer

> The Atomizer automatically splits tasks so that they can be executed in parallel.

Reviewer: Atomizer is an Nx Cloud product.

> Atomizer is natively supported by [Nx Agents](#nx-agents), providing you detailed task information in the UI, including interpreted logs and artifacts, automatically grouped by project and workflow.

Reviewer: ```suggestion
Atomizer automatically splits tasks on [Nx Agents](#nx-agents), providing you detailed task information in the UI, including interpreted logs and artifacts, automatically grouped by project and workflow.
```

> See: [Executors and Configurations](/concepts/executors-and-configurations)

### Flaky Tasks

> Tasks that will sometimes succeed and sometimes fail without any change to the inputs. These tasks are particularly problematic in CI.

Reviewer: Most commonly found during E2E tests.

> See: [Executors and Configurations](/concepts/executors-and-configurations)

### Flaky Tasks

> Tasks that will sometimes succeed and sometimes fail without any change to the inputs. These tasks are often e2e tests and are particularly problematic in CI.

Reviewer: Flaky tasks detection & retries are Nx Cloud features.

A [project](#project) that is located in a sub-folder of another project. This was made possible in Nx 15.3.

### Nx Agents

> A paid service that hosts and manages customizable worker machines for use with [distributed task execution](#distributed-task-execution).

Reviewer: At its essence, Nx Agents is a service to offshoring/externalizing your CI compute to Nx Cloud to make it more performant at lower cost. You do not necessarily need to use DTE for that, but it does DTE for you automatically when possible.

> See: [Publishable and Buildable Nx Libraries](/concepts/more-concepts/buildable-and-publishable-libraries)

### Remote Cache

A [cache](#cache) that can be shared between all developers using the repo and the CI system. This product is called [Nx Replay](#nx-replay).

> See: [Share Your Cache](/ci/features/remote-cache)

### Root-Level Project

A [project](#project) that has its own root folder at the root of the repository. Every other project in the repo is a [nested project](#nested-project).

### Run

Runs in Nx Cloud are Nx commands that result in some number of `Task`s being performed.

- `nx run my-app:build` is a run that performs a single task
- `nx run-many -t build` is a run that performs many tasks.

> - `nx affected -t test` is a run that performs zero to many tasks.

Reviewer 2: Might be helpful to use `nx affected -t test,build` here to demonstrate how it could contain many targets

When a run exists in a CI environment, Nx Cloud will automatically associate it

> with the appropriate `Run Group` and `CI Pipeline Execution`

Reviewer: Not to be confused: "Nx runs" are not "CI runs".
